compiler/executable/match_/planner/plan.rs

```diff
@@ -1483,9 +1497,13 @@ impl ConjunctionPlan<'_> {
                     match_builder.position_mapping(),
                     variable_registry,
                 );
-                let variable_positions = negation.index.clone(); // FIXME needless clone
+                let variable_positions: HashMap<Variable, ExecutorVariable> = negation
+                    .index
+                    .iter()
+                    .filter_map(|(k, v)| match_builder.current_outputs.get(k).map(|_| (k.clone(), v.clone())))
+                    .collect();
```

Comment:
Fix for the negation returning locals. I couldn't get `cloned()` to work :/
else this could have been `.filter(|(k, _)| match_builder.current_outputs.contains_key(k)).cloned()`